sentry-core/src/main/java/io/sentry/core/cache/SessionCache.java

```diff
@@ -96,6 +96,9 @@ public void store(final @NotNull SentryEnvelope envelope, final @Nullable Object
     }
 
     if (hint instanceof SessionStart) {
+
+      // TODO: should we move this to AppLifecycleIntegration? and do on SDK init? but it's too much
```

Comment:
I'd rather not do it or we find a different way to improve things.

Reply:
fwiw on sentry-cocoa it's done on the thread calling `init` (expected to be the UI thread).
Agree it'd be best to avoid work on the UI thread but we need to make sure we don't race through the current session file which `startSession` would touch upon being called

Reply:
if I do that before even calling the Lifeycicle callback, no racing.
But sentry-android does more things, like the crash marker from sentry-native with a timestamp, so a hard crash can lead to a few IOs more on the main-thread and at this point, I'd say that we are doing quite a bit.

Reply:
Sure thing, your call